### PR TITLE
feature(store): add a wriatable path option

### LINKED ISSUE #277: Store the certificates and private keys
Store the certificates and private key on the SDK instead of generating them on every connection.

At every connection, we should:
1. Check if we have a certificate stored, otherwise request one
2. Store the certificate in a writable location
3. Check if the stored certificate is still valid

This should be done at every reconnection. This way if an error occurred, we will check if it was caused by the certificate expiring.

Some points to still consider are:

- Secure store for the private key and certificate
- The possibility to use the existing structures, like the SQLite database used for the property cache, to store the certificate.

### PR DESCRIPTION
**Pull request by joshuachp**
Add a storable path for the SDK. This let's us implement the following:

- Storing the credential secret and using a paring token for authentication
- Reusing a created SSL certificate for the device

Closes #277

### CODE REVIEW

**Review by rgallor — CHANGES_REQUESTED** [review]

File: src/transport/mqtt/config/connection.rs [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+            let (bundle, certificate) = self.create_certificate(client).await?;
+
+            return ClientAuth::try_from_pem_cert(certificate, bundle.private_key)
+                .map_err(PairingError::InvalidCredentials);
+        };
+
+        let certificate_file = store_dir.join(CERTIFICATE_FILE);
+        let private_key_file = store_dir.join(PRIVATE_KEY_FILE);
+
+        self.create_and_store(client, &certificate_file, &private_key_file)
```
[the remark this answers is not in the document]
Comment (joshuachp) [reply src/transport/mqtt/config/connection.rs:113]: It was to better separate the function, at line 103 we check that we have the store directory. If we don't, we create a new one and return  it immediatelly, otherwhise we call the create and store

[the remark this answers is not in the document]
Comment (joshuachp) [reply src/transport/mqtt/config/connection.rs:113]: I will try to separate the function to make this more clear, if you can please tell me what you think about the new structure

File: src/transport/mqtt/config/connection.rs [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+
+        let certificate = client.create_certificate(&bundle.csr).await?;
+
+        debug!("credentials created");
+
+        Ok((bundle, certificate))
+    }
+
+    /// Creates and stores a new credential
+    async fn create_and_store(
```
[the remark this answers is not in the document]
Comment (joshuachp) [reply src/transport/mqtt/config/connection.rs:71]: Agread, doing the refactor as commented above 👍🏻

File: src/transport/mqtt/config/connection.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (harlem88) [reply src/transport/mqtt/config/connection.rs:113]: ok I agree with the new structure

**Review by rgallor — APPROVED** [review]

**Pull request merged**